### PR TITLE
WT-2610 Reduce hazard pointer array size.

### PR DESCRIPTION
**Pull request by sueloverso**
@michaelcahill Here are the changes.  I committed the changes to wtperf to add `session_count_idle` as a separate commit in case we don't want to keep that.  Please review.

### CODE REVIEW

**Comment by daveh86** [comment]
retest this please

**Comment by sueloverso** [comment]
@agorrod If you want to, please review these changes.  We can consider the wtperf changes separately from the hazard array changes.

**Comment by sueloverso** [comment]
@agorrod I separately committed the couple of changes we discussed.  I believe this is now ready for review.

File: src/btree/bt_discard.c [hunk 1]
```diff
@@ -63,30 +63,6 @@ __wt_page_out(WT_SESSION_IMPL *session, WT_PAGE **pagep)
...
-		__wt_sleep(0, 10000);
-	}
-	if (hp != NULL)
-		__wt_errx(session,
-		    "discarded page has hazard pointer: (%p: %s, line %d)",
-		    hp->page, hp->file, hp->line);
-	WT_ASSERT(session, hp == NULL);
-	}
-#endif
-
```
Comment (michaelcahill) [inline src/btree/bt_discard.c]: @sueloverso, what was the thinking with this change?  This check has caught real bugs in the past: is it necessary to get rid of it?

Comment (sueloverso) [inline src/btree/bt_discard.c]: @keithbostic and I discussed this a bit on the bus ride last week, questioning if it was still useful.  This diagnostic assertion was added almost 3 years ago and hasn't found anything in a while and the hazard code seems stable.  It certainly impacts diag runs.  But, it is under diagnostic mode so it doesn't impact end users.  I can go either way on this.

**Comment by agorrod** [comment]
Thanks @sueloverso lgtm - I'll go ahead and merge. Let's watch the perf tests for improvements.

**Pull request merged**